src/components/Avatar/Avatar.tsx

```diff
@@ -21,16 +21,24 @@ import { SuspenseImg } from "../../utils/SuspenseImg";
 import styles from "./Avatar.module.css";
 import { useIdColorHash } from "./useIdColorHash";
 
-type AvatarProps = JSX.IntrinsicElements["span"] & {
+type AvatarProps = (
+  | JSX.IntrinsicElements["button"]
+  | JSX.IntrinsicElements["span"]
+) & {
+  as: "button" | "span";
   src?: React.ComponentProps<typeof SuspenseImg>["src"];
   id: string;
   name: string;
   type?: "square" | "round";
   size?: CSSStyleDeclaration["height"];
+  onClick: (e: React.MouseEvent) => void;
   onError?: React.ComponentProps<typeof SuspenseImg>["onError"];
 };
 
-export const Avatar = forwardRef<HTMLSpanElement, AvatarProps>(function Avatar(
+export const Avatar = forwardRef<
+  HTMLSpanElement | HTMLButtonElement,
```

Comment:
Could this please be generic depending on `onClick`'s presence or `as` so we don't need to do type assertions on the caller side

Reply:
Yeah this gets a lot trickier with forwardRef but I think bad types here will yield landmines and footguns for callers who are unfamiliar with compound.

Reply:
I think this expresses the reality of the ref, it can either be a button or a span. I can make it more generic and say `HTMLElement` if you think that brings more value 🤷

Reply:
Can you document it at least `onClick ? button : span` but in English? Is the intention for the storybook to be the developer documentation too? Compound seems to be lacking jsdoc or any documentation for what props mean. Like type=square doesn't say anything about it being a rounded square, onClick doesn't say it'll change the underlying element etc. The overall component doesn't say what its for, that it generates a fallback avatar. I think the lack of documentation puts us in a far worse place than using BaseAvatar in react-sdk

Reply:
Have added JSDoc comments, it was definitely lacking, and I will actually document all the other components with JSDoc comment to make sure this practise spreads across the codebase. Thank you for calling that out.

We would likely not ever document implementation details like "square has rounded corners". But as discussed out of band, documenting the underlying element used, can be greatly beneficial from an a11y point of view.